test/test_quantized.py

```diff
@@ -1089,420 +1089,422 @@ def test_qlinear(self, batch_size, input_channels, output_channels,
-                                                  zero_point_min=0,
-                                                  zero_point_max=0)]),
-           strideH=st.integers(1, 3), strideW=st.integers(1, 3),
-           padH=st.integers(1, 2), padW=st.integers(1, 2),
-           channelwise=st.booleans())
-    def test_qconv_unpack(self, X, strideH, strideW, padH, padW, channelwise):
-        (inputs, filters, bias, groups) = X
-        inputs, (inputs_scale, inputs_zero_point, inputs_qtype) = inputs
-        filters, (filters_scale, filters_zero_point, filters_qtype) = filters
-        bias, (bias_scale, bias_zero_point, bias_qtype) = bias
-
-        if channelwise:
-            output_channels = filters.shape[0]
-            filters_scale = torch.tensor([filters_scale] * output_channels)
-            filters_zero_point = torch.tensor([filters_zero_point] * output_channels)
-
-        qconv_prepack = torch.ops.quantized.conv_prepack
-        qconv_unpack = torch.ops.quantized.conv_unpack
-
-        W = torch.from_numpy(filters).to(torch.float)
-        if channelwise:
-            W_q = torch.quantize_per_channel(W,
-                                             scales=filters_scale,
-                                             zero_points=filters_zero_point,
-                                             axis=0,
-                                             dtype=filters_qtype)
-        else:
-            W_q = torch.quantize_per_tensor(W, scale=filters_scale, zero_point=filters_zero_point, dtype=filters_qtype)
-
-        # Pack weights using weight packing operator
-        strides = [strideH, strideW]
-        paddings = [padH, padW]
-        dilations = [1, 1]
-        bias = torch.from_numpy(bias).to(torch.float)
-        W_packed = qconv_prepack(W_q, bias, strides, paddings, dilations, groups)
-        # Unpack weights weight unpacking operator (Used for serialization)
-        W_unpacked = qconv_unpack(W_packed)[0]
-        bias = qconv_unpack(W_packed)[1]
-
-        # Assert equal
-        np.testing.assert_equal(W_q.int_repr().numpy(), W_unpacked.int_repr().numpy())
-        if channelwise:
-            np.testing.assert_array_almost_equal(np.float32(W_q.q_per_channel_scales().numpy()),
-                                                 np.float32(W_unpacked.q_per_channel_scales().numpy()),
-                                                 decimal=4)
-            np.testing.assert_equal(W_q.q_per_channel_zero_points().numpy(), W_unpacked.q_per_channel_zero_points().numpy())
-        else:
-            np.testing.assert_equal(np.float32(W_q.q_scale()), np.float32(W_unpacked.q_scale()))
-            np.testing.assert_equal(W_q.q_zero_point(), W_unpacked.q_zero_point())
+    def test_qconv(self):
+        if 'qnnpack' in torch.backends.quantized.supported_engines:
+            if not IS_WINDOWS and not IS_PPC and not TEST_WITH_UBSAN:
+                with enable_mobile_quantized_engine():
```

Comment:
I'd change it to `with override_quantized_engine("qnnpack")` and ensure that fbgemm tests are surrounded by it too.

If there are too many tests to surround - you can also look into parametrized tests or just inheritance